### PR TITLE
agdaPackages: Don't build broken packages on hydra

### PR DESCRIPTION
**Pull request by turion**
###### Motivation for this change

Follow-up to https://github.com/NixOS/nixpkgs/pull/130263/files#r674812698

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

CC @sternenseemann @cdepillabout @alexarice @neosimsim

### CODE REVIEW

File: pkgs/build-support/agda/default.nix [hunk 1]
```diff
@@ -76,7 +77,7 @@ let
           find -not \( -path ${everythingFile} -or -path ${lib.interfaceFile everythingFile} \) -and \( ${concatMapStringsSep " -or " (p: "-name '*.${p}'") (extensions ++ extraExtensions)} \) -exec cp -p --parents -t "$out" {} +
           runHook postInstall
         '';
-      };
+      } // (if broken then { hydraPlatforms = lib.platforms.none; } else {});
```
Comment (sternenseemann) [inline pkgs/build-support/agda/default.nix:80]: `broken` and `hydraPlatforms` both need to go into the `meta` set, so this change will probably have no effect at all.

Comment (turion) [reply pkgs/build-support/agda/default.nix]: Better?

**Comment by alexarice** [comment]
I admit I haven't quite been following what's been going on in the haskell-updates branch but this looks like a reasonable change

File: pkgs/build-support/agda/default.nix [hunk 2]
```diff
@@ -76,6 +77,7 @@ let
           find -not \( -path ${everythingFile} -or -path ${lib.interfaceFile everythingFile} \) -and \( ${concatMapStringsSep " -or " (p: "-name '*.${p}'") (extensions ++ extraExtensions)} \) -exec cp -p --parents -t "$out" {} +
           runHook postInstall
         '';
+        meta = if meta ? broken && meta.broken then meta // { hydraPlatforms = lib.platforms.none; } else meta;
```
Comment (sternenseemann) [inline pkgs/build-support/agda/default.nix:80]: Looks good, small suggestion which is only a refactor:

```suggestion
        meta = if meta.broken or false then meta // { hydraPlatforms = lib.platforms.none; } else meta;
```

**Review by sternenseemann — APPROVED** [review]
Okay with me if it is intentional that setting broken will overwrite any manually set `hydraPlatforms`.

**Comment by alexarice** [comment]
Maybe I'm missing something but shouldn't broken packages not be built by hydra by default?

**Comment by turion** [comment]
> Okay with me if it is intentional that setting broken will overwrite any manually set `hydraPlatforms`.

Yes, I think that's fine for now. And if we encounter such a case we can think about that again.

**Comment by turion** [comment]
> Maybe I'm missing something but shouldn't broken packages not be built by hydra by default?

@sternenseemann probably knows better. It seems that our having set some packages broken didn't prevent them from being built.

**Comment by sternenseemann** [comment]
> Maybe I'm missing something but shouldn't broken packages not be built by hydra by default?

They won't be built indeed, but `hydraPlatforms = [];` also prevents them from being evaluated which prevents error messages about the package being broken from showing up in the eval errors tab.

We have tried to keep the `haskell-updates` evaluation errors tabs as clean as possible historically since it helps a lot when actual eval errors that cause issues occur.

**Pull request merged**